src/test/java/com/google/collegeplanner/CourseTest.java

```diff
@@ -0,0 +1,115 @@
+// limitations under the License.package com.google.collegeplanner.data;
+
+package com.google.collegeplanner.data;
+
+import java.text.ParseException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests Course */
+@RunWith(JUnit4.class)
+public final class CourseTest {
+  private String[] gradingSystem;
+  private Course course;
+
+  @Before
+  public void before() {
+    gradingSystem = new String[] {"Regular", "Pass-Fail"};
+  }
+
+  /*
+   * This method tests the getSemesterSeason() method of the Class class
+   */
+  @Test
+  public void getSemesterSeasonSuccess() throws Exception {
+    course = new Course("CMSC101", "Introduction to Computer Science", "202008", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    Assert.assertEquals("Fall", course.getSemesterSeason());
+  }
+
+  /*
+   * This method tests the getSemesterYear() method of the Class class
+   */
+  @Test
+  public void getSemesterYearSuccess() throws Exception {
+    course = new Course("CMSC101", "Introduction to Computer Science", "202008", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    Assert.assertEquals(2020, course.getSemesterYear());
+  }
+
+  @Test
+  public void invalidMonthNumber() {
+    course = new Course("CMSC101", "Introduction to Computer Science", "202003", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    try {
+      course.getSemesterSeason();
+      Assert.fail();
+    } catch (ParseException e) {
+      Assert.assertEquals("Invalid Start Month", e.getMessage());
+    }
+  }
+
+  @Test
+  public void invalidMonthFormat() {
+    course = new Course("CMSC101", "Introduction to Computer Science", "2020f3", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    try {
+      course.getSemesterSeason();
+      Assert.fail();
+    } catch (ParseException e) {
+      Assert.assertEquals("Invalid Start Month Format", e.getMessage());
+    }
+  }
+
+  @Test
+  public void invalidYearNumber() {
+    course = new Course("CMSC101", "Introduction to Computer Science", "177608", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    try {
+      course.getSemesterYear();
+      Assert.fail();
+    } catch (ParseException e) {
+      Assert.assertEquals(
+          "Invalid Year. (Too far in the future or too far in the past)", e.getMessage());
+    }
+  }
+
+  @Test
+  public void invalidYearFormat() {
+    course = new Course("CMSC101", "Introduction to Computer Science", "r8e108", 4, "CMSC",
+        "Introductory class to Computer Science", gradingSystem, null, null, null, null, null);
+    try {
```

Comment:
instead of try/catch, we should be using `assertThrows`:

```
    assertThrows(
        ParseException.class,
        () -> course.getSemesterYear());
```

or to also do an expectation on the message:

```
  ParseException thrown =
      assertThrows(
          ParseException.class,
          () -> course.getSemesterYear());
  assertThat(thrown).hasMessageThat().contains("Invalid Semester Year Format");
```

if you have access, go/java-testing/exceptions has more reading on this